### ISSUE TITLE
Add Test for same-payment_hash-different-CLTV HTLC sorting

### ISSUE THREAD

**Issue by TheBlueMatt** [lightningdevkit/rust-lightning#324]
This was fixed in #319 but no test was added.

**Comment by TheBlueMatt** [comment]
If there arent some already, this should probably also come with BOLT test vectors.

**Comment by ariard** [comment]
@rrybarczyk, this one should be a good first issue if you're looking for something to hack on :)

**Comment by varunsrin** [comment]
i can pick this up, if its still relevant

**Comment by ariard** [comment]
@varunsrin Yes this one is still relevant. You should check duplicate_htlc_test, test_duplicate_htlc_different_direction_onchain, test_duplicate_payment_hash_one_failure_one_success but IIRC they don't test this general case.

Also you may have a look on https://github.com/rust-bitcoin/rust-lightning/issues/240, going through bolt 2 and checking we implement everything right you will surely find nice tests to write. And beyond learn about the core operations of LN.

(going to label "good first issue" more this week, also you can grep really easy TODOs like L773 in peer_handler or others :) )

**Comment by Harshdev098** [comment]
Hey @TheBlueMatt is this issue still exist, I want to contribute to it!

**Comment by Harshdev098** [comment]
Hey @TheBlueMatt , Can you please give some context or reference how I can get cltv from commitment tx

![Image](https://github.com/user-attachments/assets/ae549121-8265-437a-85a0-3764791db4c4)
